src/components/LoginForm.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useRef } from 'react';
+import { useGenerateRandomUser } from '../hooks/useGenerateRandomUser';
+
+export function LoginForm({ onLogin }: { onLogin: (username: string, password: string) => void }) {
+  const { status, user } = useGenerateRandomUser();
+  const formRef = useRef<HTMLFormElement>(null);
+
+  return status === 'success' ? (
```

Comment:
You are handling the success case only. Without handling the error case (e.g., when all docker containers are stopped in your local workspace) it will look like this:

![image](https://user-images.githubusercontent.com/5851088/172697354-7bb9b2b6-2c32-4e16-9885-5ef8f61810f8.png)

I would expect to see the login form, but with an empty user name and password field. It should be possible that the user can enter any username and password and still be able to login. Please test!

We should then of course replace the paragraph `A random username and password is generated for you. ...` with

> Please enter a preferred username and password. You can use the same username and password next time to continue your work. Your previous username and password are stored as a cookie for your convenience.